update.ps1

```diff
@@ -1,201 +1,532 @@
+    $account = Convert-StringToBoolean $account
+
+    # Define properties to compare for update
+    $accountPropertiesToCompare = $account.PsObject.Properties.Name
+    #endRegion account
+
+    #region Verify account reference
+    $actionMessage = "verifying account reference"
+    
+    if ([string]::IsNullOrEmpty($($actionContext.References.Account))) {
+        throw "The account reference could not be found"
+    }
+    #endregion Verify account reference
+    
+    #region Create access token
+    $actionMessage = "creating access token"
+    
+    $createAccessTokenBody = @{
+        grant_type    = "client_credentials"
+        client_id     = $actionContext.Configuration.EntraIDAppId
+        client_secret = $actionContext.Configuration.EntraIDAppSecret
+        scope         = $actionContext.Configuration.KPNMWPScope
+    }
+    
+    $createAccessTokenSplatParams = @{
+        Uri             = "https://login.microsoftonline.com/$($actionContext.Configuration.EntraIDTenantID)/oauth2/v2.0/token/"
+        Headers         = $headers
+        Method          = "POST"
+        ContentType     = "application/x-www-form-urlencoded"
+        UseBasicParsing = $true
+        Body            = $createAccessTokenBody
+        Verbose         = $false
+        ErrorAction     = "Stop"
+    }
+    
+    $createAccessTokenResonse = Invoke-RestMethod @createAccessTokenSplatParams
+    
+    Write-Verbose "Created access token. Result: $($createAccessTokenResonse | ConvertTo-Json)"
+    #endregion Create access token
+    
+    #region Create headers
+    $actionMessage = "creating headers"
+    
+    $headers = @{
+        "Authorization"   = "Bearer $($createAccessTokenResonse.access_token)"
+        "Accept"          = "application/json"
+        "Content-Type"    = "application/json;charset=utf-8"
+        "Mwp-Api-Version" = "1.0"
+    }
+    
+    Write-Verbose "Created headers. Result: $($headers | ConvertTo-Json)."
+    #endregion Create headers
+
+    #region Get account
+    # API docs: https://mwpapi.kpnwerkplek.com/index.html, specific API call: GET /api/users/{identifier}
+    $actionMessage = "querying account where [$($correlationField)] = [$($correlationValue)]"
+
+    $getKPNLisaAccountSplatParams = @{
+        Uri         = "$($actionContext.Configuration.MWPApiBaseUrl)/users/$($actionContext.References.Account)"
+        Method      = "GET"
+        Body        = @{
+            select = "$($accountPropertiesToQuery -join ',')"
+        }
+        Verbose     = $false
+        ErrorAction = "Stop"
     }
-    $PreviousPerson = Invoke-RestMethod @LisaRequest @SplatParams
 
-    $outputContext.PreviousData = $PreviousPerson | Select-Object -Property ([array] $outputContext.Data.PSObject.Properties.Name)
+    Write-Verbose "SplatParams: $($getKPNLisaAccountSplatParams | ConvertTo-Json)"
+
+    # Add header after printing splat
+    $getKPNLisaAccountSplatParams['Headers'] = $headers
+
+    $getKPNLisaAccountResponse = $null
```

Comment:
Misschien kunnen we deze variable verwijderen en -in plaats van- alleen de `$correlatedAccount` laten staan. Dat scheelt weer een onnodige assignment.

Reply:
In theorie kan dat zeker, maar het voegt niks extra toe aan de connector nu, dus voor nu laat ik het zo.